### DISCUSSION TITLE
【请教】如何动态修改固态层颜色？

### DISCUSSION THREAD

**Discussion by HankeAvailable** [Tencent/libpag#1972] [General] [answered]
看到官方有说可以动态修改固态层颜色的feature，也看到了对应的接口
 void setSolidColor(const Color& value);

请问如何获取到的pag资产里面的固态层呢，是和lottie一样通过图层名称吗，但是没有看到相关的方法，以及如果我效果中有多个颜色需要更改，又是如何操作呢

平台:Android
非常感谢

**Comment by kevingpqi123** [comment, marked as answer]
建议系统看下 pag 的 API 接口，关于图层的获取，可以看下 PAGComposition 和 PAGFile 中的相关方法（getLayers 相关），无论是通过 LayerName 、图层类型、index 获取都是支持的

**Reply by hubeishuaige** [reply]
老哥，有没有交流群呀，很多问题，群友们可以互相讨论一下嘛，这样才能更好的发展社区嘛

**Reply by HankeAvailable** [reply]
pagView.setPathAsync("assets://output.pag", pagFile -> {
            Log.i("pag layer number",""+pagFile.numImages());
            PAGLayer[] array = pagFile.getLayersByName("customLayer");
            ((PAGSolidLayer)(array[0])).setSolidColor(0xff0000);

        });

**Reply by HankeAvailable** [reply]
自问自答系列，不过还是建议有对应的交流群
